netket/utils/_dependencies_check.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2020, 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This file contains logic to check version of NetKet's dependencies and hard-error in 
```

Comment:
can't we enforce the version in pip? what is the issue there?

Reply:
If you have a recent pip version (~from the last 12 months), there is no issue, and pip will take care of installing an older version of optax.
If you have an older pip version, pip ignores compatibility bounds on dependencies when you update an already installed package.

Thinking about what happened to you the other day: you had older pip, and updating netket would not have updated optax. Then again, old pip is not able to install jax/flax either so maybe it's not an issue.

This code triggering an error here is just to be extra-safe. 
New pip should be sufficient, but 'new pip' is fairly new so it's not easy to assume everyone has it...

@femtobit what do you think?

Reply:
I'd merge this as is, anyhow this does not create problems at all.
In an year from now, hopefully people will have more recent pips and then we can drop this.